src/Microsoft.Azure.NotificationHubs/KeyValueConfigurationManager.cs

```diff
@@ -106,10 +122,208 @@ private static NameValueCollection CreateNameValueCollectionFromConnectionString
+        }
+
+        internal TokenProvider CreateTokenProvider()
+        {
+            IList<Uri> endpointAddresses = GetEndpointAddresses(connectionProperties["StsEndpoint"], (string)null);
+            string connectionProperty1 = connectionProperties["SharedSecretIssuer"];
+            string connectionProperty2 = connectionProperties["SharedSecretValue"];
+            string connectionProperty3 = connectionProperties["SharedAccessKeyName"];
+            string connectionProperty4 = connectionProperties["SharedAccessKey"];
+            string connectionProperty5 = connectionProperties["WindowsDomain"];
+            string connectionProperty6 = connectionProperties["WindowsUsername"];
+            SecureString windowsPassword1 = GetWindowsPassword();
+            string connectionProperty7 = connectionProperties["OAuthDomain"];
+            string connectionProperty8 = connectionProperties["OAuthUsername"];
+            SecureString oauthPassword1 = GetOAuthPassword();
+            string issuerName = connectionProperty1;
+            string issuerKey = connectionProperty2;
+            string sharedAccessKeyName = connectionProperty3;
+            string sharedAccessKey = connectionProperty4;
+            string windowsDomain = connectionProperty5;
+            string windowsUser = connectionProperty6;
+            SecureString windowsPassword2 = windowsPassword1;
+            string oauthDomain = connectionProperty7;
+            string oauthUser = connectionProperty8;
+            SecureString oauthPassword2 = oauthPassword1;
+            return CreateTokenProvider((IEnumerable<Uri>)endpointAddresses,
+                issuerName,
+                issuerKey,
+                sharedAccessKeyName,
+                sharedAccessKey,
+                windowsDomain,
+                windowsUser,
+                windowsPassword2,
+                oauthDomain,
+                oauthUser,
+                oauthPassword2);
+        }
+
+        private static TokenProvider CreateTokenProvider(
+            IEnumerable<Uri> stsEndpoints,
+            string issuerName,
+            string issuerKey,
+            string sharedAccessKeyName,
+            string sharedAccessKey,
+            string windowsDomain,
+            string windowsUser,
+            SecureString windowsPassword,
+            string oauthDomain,
+            string oauthUser,
+            SecureString oauthPassword)
+        {
+            if (!string.IsNullOrWhiteSpace(sharedAccessKey))
+            {
+                return TokenProvider.CreateSharedAccessSignatureTokenProvider(sharedAccessKeyName, sharedAccessKey);
+            }
+            if (string.IsNullOrWhiteSpace(issuerName))
+            {
+                int num = stsEndpoints == null ? 0 : (stsEndpoints.Any<Uri>() ? 1 : 0);
+                bool flag1 = !string.IsNullOrWhiteSpace(windowsUser) && windowsPassword != null && windowsPassword.Length > 0;
```

Comment:
flag1 and flag2 don't really mean much in this context. Could they be named hasWindowsPassword and has oAuthPassword?

Reply:
renamed to `hasWindowsPassword` and `hasOAuthPassword`